Source/SwiftLintFramework/Models/Location.swift

```diff
@@ -60,9 +60,9 @@ public struct Location: CustomStringConvertible, Comparable, Codable {
     /// - parameter position: The absolute position returned from SwiftSyntax.
     public init(file: SwiftLintFile, position: AbsolutePosition) {
         self.file = file.path
-        if let sourceLocation = file.locationConverter?.location(for: position) {
-            line = sourceLocation.line
-            character = sourceLocation.column
+        if let lineAndCharacter = file.stringView.lineAndCharacter(forByteOffset: ByteCount(position.utf8Offset)) {
```

Comment:
@marcelofabri why this change? Did you benchmark it as being faster?

Reply:
And why not delegate to this init? https://github.com/realm/SwiftLint/blob/79347a17286476509ff79cf92f959e42597d4b56/Source/SwiftLintFramework/Models/Location.swift#L45

Reply:
I didn’t benchmark it, but the existing code was giving wrong column numbers for lines with UTF-8 characters. The added example was one case.

Reply:
(I should’ve delegated it to the other init though)

Reply:
Giving our caching of `StringView`s, it may actually be faster and more accurate than SwiftSyntax.

I'd encourage you to benchmark it in SwiftLint and to file a bug report towards SwiftSyntax if there's a correctness bug.

Reply:
> (I should’ve delegated it to the other init though)

Good opportunity for a follow-up PR 😄

Reply:
I’ll make that change and open an issue with SwiftSyntax later today 🙏

Reply:
I think we were using this wrong. From `SourceLocation`:

```
/// The UTF-8 byte offset from the beginning of the line where this location
  /// resides. 1-based.
  public var column: Int? {
    return compLoc?.column
  }
```

As far as I can tell, we expect `Location`'s `character` to be a UTF-16 offset, so this would explain the behavior I was seeing. Does this make sense?

Reply:
Hmmm, I expect it to be the logical character offset on the line so that Xcode renders it at the correct location.